platforms/enterprise/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/legacy/BuildScanScopeIdsIntegrationTest.groovy

```diff
@@ -42,5 +42,7 @@ class BuildScanScopeIdsIntegrationTest extends AbstractIntegrationSpec {
         output.contains "ids: [buildInvocation: $buildInvocationId, workspace: $workspaceId, user: $userId]"
     }
 
-
+    def setup() {
+        executer.withEagerClassLoaderCreationCheckDisabled()
```

Comment:
We can't make strict classloading the default because of backwards compatibility, but it would be great to do so.
Our features requiring non-strict classloading is a smell.
Now with this change we are preventing our future selves to do get away with it.

❓  Maybe we need a dedicated classloader scope for this whole feature in order to isolate its lifecycle and classloading. I feel like this is important enough to be discussed before committing to the current implementation.

Reply:
I updated the error message here with the current failure (when this setup method is removed).  

I would like to understand this test better.

As I understand it, strict vs. non-strict classloading behavior should make no difference to how `InitBuild` works.  Even if this class loader scope is **not** locked, the `InitProjectSource`s we're looking for must already be available on its classpath here.  If some other classes aren't available, because they are loaded lazily, that shouldn't be a problem, because this scope is only used in a limited way.  This test is (correctly) reporting it could be a problem here, so the check must be disabled.  

So I don't believe this behavior will actually be a problem with strict classloading, but I'm far from an expert.